app/components/dsfr/alert_component.rb

```diff
@@ -13,20 +13,25 @@ def prefix_for_state
   end
 
   def alert_class(state)
-    ["fr-alert fr-alert--#{state}", extra_class_names].compact.flatten
+    if size == 'small'
+      ["fr-alert fr-alert--sm fr-alert--#{state}", extra_class_names].compact.flatten
+    else
+      ["fr-alert fr-alert--#{state}", extra_class_names].compact.flatten
+    end
```

Comment:
tu peux éviter la duplication avec : 

```suggestion
class_names(
  "fr-alert fr-alert--#{state}" => true,
  "fr-alert--sm" => size == 'small',
  extra_class_names => true
 )
```